hathor/verification/verification_service.py

```diff
@@ -68,9 +68,11 @@ def validate_full(
 
         If no exception is raised, the ValidationState will end up as `FULL` or `CHECKPOINT_FULL` and return `True`.
         """
+        assert self._tx_storage is not None
         from hathor.transaction.transaction_metadata import ValidationState
 
         meta = vertex.get_metadata()
+        vertex.init_static_metadata_from_storage(self._settings, self._tx_storage)
```

Comment:
It could be done by the `VertexHandler.on_new_vertex()` just before it calls verification, but they're more or less equivalent. When a new vertex is received, the verification is the first step to need static metadata, so I think it makes sense to defer this calculation until we actually need it.

Also, there's the added benefit that existing tests that call verification will get this call for free. If the call was made before, tests would have to do it manually before, too.

Reply:
What happens if the static metadata has already been initialized and this method is called again?

Reply:
This method calls `set_static_metadata()`, which is:

```python
    def set_static_metadata(self, static_metadata: StaticMetadataT | None) -> None:
        """Set this vertex's static metadata. After it's set, it can only be set again to the same value."""
        assert not self._static_metadata or self._static_metadata == static_metadata, (
            'trying to set static metadata with different values'
        )
        self._static_metadata = static_metadata
```

I think it would be even better to disallow it completely with just `assert not self._static_metadata`, not permitting calls even if the value is the same. For normal full node operation, this is indeed never called more than once. But for tests, we have a lot of places that need the the static metadata before calling `on_new_vertex()`. So I think in a future PR we could improve all those tests and then remove the equality assertion.

Reply:
Should we log a warning (or maybe info/debug message) if the static metadata already exists and is equal just to have an idea if this is happening an in what cases?

Reply:
Sure, I think it makes sense. I added it as a warning so we get alerted if this happens, which it shouldn't. Then, in the future, I'll update tests and prohibit this. Done in https://github.com/HathorNetwork/hathor-core/pull/1014/commits/6866b723f902b7960e46df61bb33fd08c0ab1839

Reply:
Thanks!